### DISCUSSION TITLE
Not getting usage data

### DISCUSSION THREAD

**Discussion by chaffinched** [BottlecapDave/HomeAssistant-OctopusEnergy#906] [Q&A] [answered]
Hi, I've got a fresh install of Home Assistant, HACS and your package, all done today. 
HAOS: 2024.6.1
HACS: 1.34.0
HA Octopus: 11.1.0

After setting the account ID and API key and ticking the box for the Home Mini, I can see 2 devices (gas and electric, both correctly identified) and 16 entities, listed below. What I don't see is usage data, or the sensors that I would expect to provide that data. The only sensor that I'm seeing data for is "Off Peak Electricity", and I'm not on an off peak / Economy 7 tariff.

If I run the following, I get valid data for yesterday, although this has been intermittent, which is why they sent me the Home Mini, so I could get live data.
curl -u "sk_live_REDACTED:" "https://api.octopus.energy/v1/electricity-meter-points/METERID_REDACTED/meters/METERID_REDACTED/consumption/"

Debugging in HA shows two interesting things. 

First, these two queries appear every 60 seconds, which seems excessive for the data they are fetching:
```
2024-06-11 13:38:15.526 DEBUG (MainThread) [custom_components.octopus_energy.coordinators.electricity_rates] Finished fetching ELECTRICITY_RATES_METERID_REDACTED data in 5.812 seconds (success: True)
2024-06-11 13:39:06.392 DEBUG (MainThread) [custom_components.octopus_energy.coordinators.saving_sessions] Finished fetching saving_sessions_A-B84324C2 data in 0.000 seconds (success: True)
```

And this, which is odd, because I think I am on a standard rate, not a day / night one:
`2024-06-11 13:38:15.523 WARNING (MainThread) [custom_components.octopus_energy.api_client] Failed to send request (https://api.octopus.energy/v1/products/VAR-22-11-01/electricity-tariffs/E-FLAT2R-VAR-22-11-01-H/standard-unit-rates?period_from=2024-06-09T23:00:00Z&period_to=2024-06-12T23:00:00Z&page=1): 400; {"detail":"This tariff has day and night rates, not standard."}`

Have I missed a step in the setup, do I need to manually create the sensors, or is Octopus simply not returning the data you are expecting?

Octopus Energy entities:
binary_sensor.octopus_energy_ACCOUNTID_REDACTED_greenness_forecast_highlighted
binary_sensor.octopus_energy_ACCOUNTID_REDACTED_octoplus_saving_sessions
binary_sensor.octopus_energy_electricity_METERID_REDACTED_off_peak
event.octoplus_saving_session_events_ACCOUNTID_REDACTED
event.octopus_energy_electricity_METERID_REDACTED_current_day_rates
event.octopus_energy_electricity_METERID_REDACTED_next_day_rates
event.octopus_energy_electricity_METERID_REDACTED_previous_consumption_override_rates
event.octopus_energy_electricity_METERID_REDACTED_previous_consumption_rates
event.octopus_energy_electricity_METERID_REDACTED_previous_day_rates
event.octopus_energy_gas_METERID_REDACTED_current_day_rates
event.octopus_energy_gas_METERID_REDACTED_next_day_rates
event.octopus_energy_gas_METERID_REDACTED_previous_consumption_override_rates
event.octopus_energy_gas_METERID_REDACTED_previous_consumption_rates
event.octopus_energy_gas_METERID_REDACTED_previous_day_rates
text.octopus_energy_electricity_METERID_REDACTED_previous_accumulative_cost_override_tariff
text.octopus_energy_gas_METERID_REDACTED_previous_accumulative_cost_override_tariff

**Comment by BottlecapDave** [comment]
Hello and sorry that you're having issues. The integration automatically creates various sensors based on your tariff and available data. The tariff that is being reported doesnt look right. Could you please follow the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq/#ive-been-asked-for-my-meter-information-in-a-bug-request-how-do-i-obtain-this) to download your diagnostics and provide them here?

**Reply by chaffinched** [reply]
If I try getting the diags from the following location, nothing happens. I've tried 2 different browsers without extensions / add-ons:
`Settings > Devices > Integrations > Octopus Energy > 3 dots > Download diags`

But if I go to Devices, the meter provides me with the data in octopus_energy-Electricity Meter.json

I've also attached the HA  logs.
[octopus_energy-Electricity Meter.json](https://github.com/user-attachments/files/15801615/octopus_energy-Electricity.Meter.json)
[home-assistant.log](https://github.com/user-attachments/files/15801620/home-assistant.log)

**Comment by BottlecapDave** [comment, marked as answer]
This should now be resolved as part of v11.1.1

**Reply by chaffinched** [reply]
Wonderful, thanks!

I've gone from 16 entities to 52, of which 49 are getting data. "Previous Rate Gas", "Next Rate Gas" and "Current Consumption Gas" are all giving "Unknown", possibly because I haven't used any gas today.

**Reply by BottlecapDave** [reply]
Previous and next rates will be unknown if they're the same value as current (as per the docs)

Current gas consumption will be populated if it's populated in the app and as you mentioned might be related to not using any.
